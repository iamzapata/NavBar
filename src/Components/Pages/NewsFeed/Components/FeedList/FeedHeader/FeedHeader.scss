@import "styles/variables.scss";

.FeedHeader {
  margin: 20px 0;
  .title {
    color: $color-black-text;
  }
  .subtitle {
    color: $color-gray-content-text;
  }
}
