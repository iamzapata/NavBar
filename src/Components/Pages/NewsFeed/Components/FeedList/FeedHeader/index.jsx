import React from "react"
import { distanceInWordsToNow } from "date-fns"
import "./FeedHeader.scss"

const FeedHeader = ({ feed: { title, author, updated_at } }) => (
  <div className="FeedHeader">
    <p className="title has-text-weight-normal is-3">{title}</p>
    <p className="subtitle is-6">
      By {author.name} &middot;
      {distanceInWordsToNow(updated_at)} ago
    </p>
  </div>
)

export default FeedHeader
