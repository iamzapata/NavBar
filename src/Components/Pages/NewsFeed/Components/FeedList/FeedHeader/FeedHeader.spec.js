import React from "react"
import { render, cleanup } from "react-testing-library"
import FeedHeader from "./index"

const props = { title: "", updated_at: "", author: { name: "" } }

describe("<FeedHeader />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(<FeedHeader feed={props} />)
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
