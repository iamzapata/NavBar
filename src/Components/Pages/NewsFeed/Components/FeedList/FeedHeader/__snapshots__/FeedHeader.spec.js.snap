// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedHeader /> Should render self 1`] = `
<div
  class="FeedHeader"
>
  <p
    class="title has-text-weight-normal is-3"
  />
  <p
    class="subtitle is-6"
  >
    By
      
    <span
      class="Separator"
    >
      Â·
    </span>
    almost NaN years
     ago 
  </p>
</div>
`;
